src/cls/rgw/cls_rgw.cc

```diff
@@ -990,6 +990,42 @@ static int read_key_entry(cls_method_context_t hctx, cls_rgw_obj_key& key,
   return 0;
 }
 
+static int complete_cleanup_obj(cls_method_context_t hctx,
```

Comment:
I would love to see a high-level comment describing the purpose/role of this function